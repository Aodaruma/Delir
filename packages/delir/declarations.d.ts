declare module '*.styl' {
    const _: { [className: string]: string }
    export = _
}

declare module '*.json' {
    const _: { [key: string]: any }
    export = _
}

declare module 'devtron' {
    export function install(): void
}

declare module 'mouse-wheel' {
    const _: (element: Element, callback: (dx: number, dy: number) => void, noScroll?: boolean) => void

    export = _
}

declare module 'electron-devtools-installer' {
    type Extensions = 'REACT_DEVELOPER_TOOLS'
    export const REACT_DEVELOPER_TOOLS = 'REACT_DEVELOPER_TOOLS'
    export default function installExtension(extension: Extensions): Promise<void>
}

declare module 'parse-color' {
    function parseColor(
        colorCode: string,
    ): {
        /** RGB values */
        rgb: [number, number, number]
        /** RGBA values */
        rgba: [number, number, number, number]
        /** CSS color keyword */
        keyword: string
        /** HEX color code */
        hex: string
    }

    namespace parseColor {}
    export = parseColor
}

declare module 'form-serialize' {
    interface SerializeOption {
        /** if true, the hash serializer will be used for serializer option */
        hash?: boolean
        /** override the default serializer (hash or url-encoding) */
        serializer?: Function
        /** if true, disabled fields will also be serialized */
        disabled?: boolean
        /** if true, empty fields will also be serialized */
        empty?: boolean
    }

    function serialize(form: HTMLFormElement, options?: SerializeOption): { [name: string]: string | string[] } | string
    function serialize(
        form: HTMLFormElement,
        options?: SerializeOption & { hash: true },
    ): { [name: string]: string | string[] }

    namespace serialize {}
    export = serialize
}

declare interface SVGGElement {
    dataset: DOMStringMap
}

declare interface PointerEvent {
    path: EventTarget[]
}

declare interface Window {
    requestIdleCallback: (
        callback: (e: { timeRemaining: number; didTimeout: boolean }) => void,
        options?: { timeout: number },
    ) => void
}

declare interface Document {
    exitPointerLock(): void
}

declare interface Element {
    requestPointerLock(): void
}

declare interface HTMLInputElement {
    onpointerlockerror: (event: ErrorEvent) => void
}

declare const __DEV__: boolean

declare interface Window {
    delir: any
}
