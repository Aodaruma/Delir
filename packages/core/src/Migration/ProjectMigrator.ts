import _ from 'lodash'
import { Project } from '../Entity/Project'
import { walkAssets } from './MigrationHelper'

export default {
  isMigratable: (project: Project) => {
    if (project.formatVersion === '2017091401') {
      return true
    }

    return false
  },

  /** Migrate project to latest schema at migratable version */
  migrate: (project: Project) => {
    if (project.formatVersion === '2017091401') {
      walkAssets(project, asset => {
        asset.path = `file://${asset.path}`
      })
    }

    return project
  },
}
