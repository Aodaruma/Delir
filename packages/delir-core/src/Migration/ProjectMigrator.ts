import * as _ from 'lodash'

export default {
    isMigratable: (project: any) => {
        return false
    },

    /** Migrate project to latest schema at migratable version */
    migrate: (project: any) => {
        return project
    },
}
