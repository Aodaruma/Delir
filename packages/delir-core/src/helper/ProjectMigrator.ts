import { ClipScheme } from '../project/scheme/clip'
import { KeyframeConfigScheme } from '../project/scheme/keyframe'
import { ProjectScheme } from '../project/scheme/project'

import * as _ from 'lodash'
import toJSON from './toJSON'

const migrate000to2017091401 = (project: ProjectScheme): ProjectScheme => {
    const transformValueSchema = (kfConf: KeyframeConfigScheme) => {
        if (!_.isObject(kfConf.value)) return
        else {
            if (_.has(kfConf.value, 'assetId')) {
                kfConf.value = { type: 'asset', value: { assetId: (kfConf.value as any).assetId } }
            } else if (!_.has(kfConf.value, '_alpha') && _.has(kfConf.value, 'red')) {
                kfConf.value = {
                    type: 'color-rgb',
                    value: {
                        red: (kfConf.value as any)._red,
                        green: (kfConf.value as any)._green,
                        blue: (kfConf.value as any)._blue,
                    }
                }
            } else if (_.has(kfConf.value, '_alpha')) {
                kfConf.value = {
                    type: 'color-rgba',
                    value: {
                        red: (kfConf.value as any)._red,
                        green: (kfConf.value as any)._green,
                        blue: (kfConf.value as any)._blue,
                        alpha: (kfConf.value as any)._alpha,
                    }
                }
            }
        }
    }

    // Update keyframe value schema (Post effect is not implemented in v0.0.0)
    for (const comp of project.compositions) {
        comp.config.backgroundColor = {
            red: (comp.config.backgroundColor as any)._red,
            green: (comp.config.backgroundColor as any)._green,
            blue: (comp.config.backgroundColor as any)._blue,
        }

        for (const layer of comp.layers) {
            for (const clip of layer.clips) {
                for (const kfName of Object.keys(clip.keyframes)) {
                    for (const kf of clip.keyframes[kfName]) {
                        transformValueSchema(kf.config)
                    }
                }
            }
        }
    }

    project.formatVersion = '2017091401'
    return project
}

export default {
    isMigratable: (project: ProjectScheme) => {
        return project.formatVersion === '0.0.0'
    },

    migrate: (project: ProjectScheme) => {
        project = toJSON(project)

        if (project.formatVersion === '0.0.0') {
            project = migrate000to2017091401(project)
        }

        if (project.formatVersion === '2017091401') {
            return project
        }

        throw new Error(`Unknown schema version ${project.formatVersion}`)
    }
}
