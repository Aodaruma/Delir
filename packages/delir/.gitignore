/tmp
/prebuild
/node_modules
