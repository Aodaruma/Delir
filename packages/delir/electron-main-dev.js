const { BrowserWindow, app } = require('electron');
const __DEV__ = process.env.NODE_ENV === 'development';


let mainWindow

app.on('ready', () => {
    mainWindow = new BrowserWindow({ width: 1920, height: 1080 });
    mainWindow.loadURL(__DEV__ ? 'http://localhost:3000' : `file://${__dirname}/prebuild/index.html`)
})
