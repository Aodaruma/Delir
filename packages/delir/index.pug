doctype html
html
    head
        meta(charset="UTF-8")
        title Delir
        script(src="main.js")
    body.theme--default
        #loading Loading...
        #root
            ._container
                .view-navigation
                    ul.window-nav
                        li.window-nav-item.window-nav--close
                        //- li.window-nav-item.window-nav--minimize
                        //- li.window-nav-item.window-nav--maximize
