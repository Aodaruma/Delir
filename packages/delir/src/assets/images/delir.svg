<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100"><defs><style>.cls-1{fill:#9711cd;}.cls-2{fill:#3c3e42;}.cls-3{fill:none;}</style></defs><g><g><polygon class="cls-1" points="48.34 0 54.54 67.75 36.87 100 72.51 72.68 48.34 0"/><polygon class="cls-2" points="47.99 16.98 25.79 72.95 64.19 100 44.78 67.89 47.99 16.98"/><rect class="cls-3" width="100" height="100"/></g></g></svg>
