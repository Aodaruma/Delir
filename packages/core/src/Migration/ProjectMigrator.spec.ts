import * as p2017091401 from '../../spec/fixture/project/2017091401.delir.json'
import { deserializeProject } from '../Exporter'
import ProjectMigrator from './ProjectMigrator'

describe('ProjectMigrator', () => {
  it('Migrate to 2019052601 from 2017091401', () => {
    const beforeMigration = deserializeProject(p2017091401)
    const afterMigration = ProjectMigrator.migrate(beforeMigration)
    expect(afterMigration.assets).toMatchSnapshot()
  })
})
