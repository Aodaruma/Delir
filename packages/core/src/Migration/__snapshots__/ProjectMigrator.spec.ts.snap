// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectMigrator Migrate to 2019052601 from 2017091401 1`] = `
Array [
  Asset {
    "fileType": "mp4",
    "id": "78262d20-af63-4520-abfd-34e85e855618",
    "name": "BigBuckBunny",
    "path": "file:///Users/ragg/workspace/delir/packages/delir/utils/Dev/ExampleProject1/big_buck_bunny.mp4",
  },
  Asset {
    "fileType": "mp3",
    "id": "3748e817-9c43-43d4-be63-b8ddcd25a2fc",
    "name": "Audio",
    "path": "file:///Users/ragg/workspace/delir/packages/delir/utils/Dev/ExampleProject1/audio.mp3",
  },
  Asset {
    "fileType": "png",
    "id": "0bbb7142-e2c2-4a8a-9a98-4adc833bcff1",
    "name": "Image",
    "path": "file:///Users/ragg/workspace/delir/packages/delir/utils/Dev/ExampleProject1/image.png",
  },
]
`;
